packages/webcomponents/src/components/business-forms/payment-provisioning/document-uploads/business-document-upload-form-step.tsx

```diff
@@ -0,0 +1,236 @@
+    if (!this.businessId) console.error(missingBusinessIdMessage);
+
+    this.formController = new FormController(businessDocumentSchema(this.allowOptionalFields));
+    this.api = Api({ authToken: this.authToken, apiOrigin: config.proxyApiOrigin });
+    this.fetchData();
+  }
+
+  componentDidLoad() {
+    this.formController.values.subscribe(values =>
+      this.documents = { ...values }
+    );
+    this.formController.errors.subscribe(
+      errors => (this.errors = { ...errors }),
+    );
+  }
+
+  inputHandler = (name: string, value: string) => {
+    this.formController.setValues({
+      ...this.formController.values.getValue(),
+      [name]: value,
+    });
+  }
+
+  private fetchData = async () => {
+    this.formLoading.emit(true);
+    try {
+      const response: IApiResponse<IBusiness> = await this.api.get(this.businessEndpoint);
+      this.documents = response.data.documents;
+    } catch (error) {
+      this.serverError.emit({ data: error, message: DocumentFormServerErrors.fetchData });
+    } finally {
+      this.formLoading.emit(false);
+    }
+  }
+
+  createDocumentRecord = async (docData: EntityDocument) => {
+    this.formLoading.emit(true);
+    const payload = docData.record_data;
+    try {
+      const response = await this.api.post(this.documentEndpoint, JSON.stringify(payload));
+      return this.handleDocRecordResponse(docData, response);
+    } catch (error) {
+      this.serverError.emit({ data: error, message: DocumentFormServerErrors.sendData });
+      return false;
+    }
+  }
+
+  handleDocRecordResponse = (docData: EntityDocument,response: any) => {
+    if (response.error) {
+      this.serverError.emit({ data: response.error, message: DocumentFormServerErrors.sendData });
+      return false;
+    } else {
+      docData.setPresignedUrl(response.data.presigned_url);
+      return true;
+    }
+  }
+
+  uploadDocument = async (docData: EntityDocument) => {
+    if (!docData.presigned_url) {
+      throw new Error('Presigned URL is not set');
+    }
+
+    const response = await fetch(docData.presigned_url, {
+      method: 'PUT',
+      body: docData.fileString,
+    })
+
+    return this.handleUploadResponse(response);
+  }
+
+  handleUploadResponse = (response: any) => {
+    if (response.error) {
+      this.serverError.emit({ data: response.error, message: DocumentFormServerErrors.sendData });
+      return false;
+    } else {
+      this.submitted.emit({ data: response, metadata: { completedStep: 'documentUpload' } });
+      return true;
+    }
+  }
+
+  storeFiles = (e: CustomEvent<FileSelectEvent>) => {
+    const fileList = Array.from(e.detail.fileList) as File[];
+    const docType = e.detail.document_type;
+    const documentList = fileList.map(file => new EntityDocument({ file, document_type: docType }, this.businessId));
+    this.documentData[docType] = documentList;
+  }
+  
+  sendData = async (onSuccess?: () => void) => {
```

Comment:
Should this also receive an `onError` callback to be called on line 130 and 134?

Reply:
In line with my other response - yes I totally agree about passing an `onError` callback to this function. I want to update this in this component, and the other form step components in another PR that will address error handling logic for the entire PaymentProvisioning component